File: src/main/java/no/nav/bidrag/beregn/saertilskudd/rest/consumer/SaertilskuddCore.java
package no.nav.bidrag.beregn.saertilskudd.rest.consumer;

import no.nav.bidrag.beregn.saertilskudd.rest.dto.http.BeregnSaertilskuddGrunnlagCore;
import no.nav.bidrag.beregn.saertilskudd.rest.dto.http.BeregnSaertilskuddResultatCore;

public class SaertilskuddCore {

  public BeregnSaertilskuddResultatCore beregnSaertilskudd(
      BeregnSaertilskuddGrunnlagCore saertilskuddGrunnlagCore) {
    return new BeregnSaertilskuddResultatCore("Hallo, " + saertilskuddGrunnlagCore.getGrunnlag());
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/saertilskudd/rest/dto/http/saertilskuddcore.kt
package no.nav.bidrag.beregn.saertilskudd.rest.dto.http

import java.time.LocalDate

// Grunnlag
data class BeregnSaertilskuddGrunnlagCore(
    val beregnDatoFra: LocalDate,
    val beregnDatoTil: LocalDate,
    val grunnlag: String
)

// Resultat
data class BeregnSaertilskuddResultatCore(
    val resultat: String
)
